src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js

```diff
@@ -37,22 +35,25 @@ angular.module('virtoCommerce.DemoSolutionFeaturesModule')
                     title: 'demoSolutionFeaturesModule.blades.customer-segment-rule-creation.title',
                     subtitle: 'demoSolutionFeaturesModule.blades.customer-segment-rule-creation.subtitle',
                     template: 'Modules/$(virtoCommerce.DemoSolutionFeaturesModule)/Scripts/blades/customerSegment-rule.tpl.html',
-                    currentEntity: blade.currentEntity
+                    originalEntity: blade.currentEntity,
+                    onSelected: function (entity) {
+                        blade.currentEntity = entity;
+                    }
                 };
                 blade.activeBladeId = ruleCreationBlade.id;
                 bladeNavigationService.showBlade(ruleCreationBlade, blade);
             };
 
-            $scope.$watch('blade.currentEntity', (data) => {
-                if (data) {
-                    $scope.totalPropertiesCount = 4;
-                    $scope.filledPropertiesCount = 0;
+        $scope.$watch('blade.currentEntity', (data) => {
```

Comment:
Here arrow function could be initialized without round brackets

Reply:
May be better to make these changes in the new customer segments module?